From da6ca7d1783667d1cdfd452e2e5ec89e10abb2dd Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 16 Nov 2023 16:38:47 +0100
Subject: [PATCH 0707/1141] drm/stm: ltdc: remove reload interrupt

The reload interrupt is not use by the driver. To avoid unnecessary call
of interrupt routine, don't enable it.

Change-Id: Iedf053fb94113dae4cc922441ee6713f6dff11f6
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/340369
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/stm/ltdc.c b/drivers/gpu/drm/stm/ltdc.c
index 0c713aa51e87..0d9a94d93b8a 100644
--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -833,11 +833,10 @@ static void ltdc_crtc_atomic_enable(struct drm_crtc *crtc,
 
 	/* Enable IRQ */
 	if (ldev->caps.crtc_rotation)
-		regmap_set_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE |
-			        IER_RRIE | IER_TERRIE | IER_FURIE);
+		regmap_set_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE | IER_TERRIE |
+				IER_FURIE);
 	else
-		regmap_set_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE |
-			        IER_RRIE | IER_TERRIE);
+		regmap_set_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE | IER_TERRIE);
 
 	/* Commit shadow registers = update planes at next vblank */
 	if (!ldev->caps.plane_reg_shadow)
@@ -864,10 +863,10 @@ static void ltdc_crtc_atomic_disable(struct drm_crtc *crtc,
 	/* disable IRQ */
 	if (ldev->caps.crtc_rotation)
 		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE |
-				  IER_RRIE | IER_TERRIE | IER_FURIE);
+				  IER_TERRIE | IER_FURIE);
 	else
 		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_FUWIE | IER_FUEIE |
-				  IER_RRIE | IER_TERRIE);
+				  IER_TERRIE);
 
 	/* immediately commit disable of layers before switching off LTDC */
 	if (!ldev->caps.plane_reg_shadow)
@@ -2387,11 +2386,11 @@ int ltdc_load(struct drm_device *ddev)
 
 	/* Disable interrupts */
 	if (ldev->caps.fifo_threshold)
-		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_RRIE | IER_FUWIE |
-				  IER_TERRIE);
+		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_FUWIE |
+				  IER_RRIE | IER_TERRIE);
 	else
-		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_RRIE | IER_FUWIE |
-				  IER_TERRIE | IER_FUEIE | IER_FURIE);
+		regmap_clear_bits(ldev->regmap, LTDC_IER, IER_LIE | IER_FUWIE |
+				  IER_RRIE | IER_TERRIE | IER_FUEIE | IER_FURIE);
 
 	DRM_DEBUG_DRIVER("ltdc hw version 0x%08x\n", ldev->caps.hw_version);
 
-- 
2.39.2

